src/test/java/com/google/research/bleth/simulator/ObserverTest.java

```diff
@@ -0,0 +1,930 @@
+    }
+
+    @Test
+    public void randomObserverSurroundedByFourDirectionsNextMoveIsToItsLocation() {
+        // ------
+        // | O  |
+        // ------
+        Board board = new Board(1, 1);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+
+        Location newLocation = randomObserver.moveTo();
+
+        assertThat(newLocation).isEqualTo(zeroOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsNextMoveIsUp() {
+        // -----
+        // |   |
+        // -----
+        // | O |
+        // -----
+        Board board = new Board(2, 1);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(oneOnZeroCoordinate);
+
+        Location newLocation = randomObserver.moveTo();
+
+        assertThat(newLocation).isEqualTo(zeroOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsNextMoveIsDown() {
+        // -----
+        // | O |
+        // -----
+        // |   |
+        // -----
+        Board board = new Board(2, 1);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+
+        Location newLocation = randomObserver.moveTo();
+
+        assertThat(newLocation).isEqualTo(oneOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsNextMoveIsRight() {
+        // ----------
+        // | O |    |
+        // ----------
+        Board board = new Board(1, 2);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+
+        Location newLocation = randomObserver.moveTo();
+
+        assertThat(newLocation).isEqualTo(zeroOnOneCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsNextMoveIsLeft() {
+        // ----------
+        // |   | O  |
+        // ----------
+        Board board = new Board(1, 2);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnOneCoordinate);
+
+        Location newLocation = randomObserver.moveTo();
+
+        assertThat(newLocation).isEqualTo(zeroOnZeroCoordinate);
+    }
+
+    @Test
+    public void moveStaticObserverStayOnItsLocation() {
+        Board board = new Board(2, 2);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer staticObserver = createStaticObserverOnLocation(zeroOnZeroCoordinate);
+
+        staticObserver.move();
+
+        assertThat(staticObserver.getLocation()).isEqualTo(zeroOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByFourDirectionsStayOnItsLocation() {
```

Comment:
This seems like a duplicate of `randomObserverSurroundedByFourDirectionsNextMoveIsToItsLocation`? (Same applies to the next few tests)

Reply:
One is for moveTo and the second is for move, but I guess that testing move is enough because it uses moveTo.